### ISSUE TITLE
FeatureMap suggestion

### ISSUE THREAD

**Issue by stevage** [Turfjs/turf#1564]
There is featureEach and featureReduce, but no featureMap. It would be useful when creating derived datasets, where you don't want to interfere with the existing data.

My particular use case involves wanting to apply bboxClip to each feature within a FeatureCollection.

**Comment by rowanwins** [comment]
Hey @stevage 

I think there's been some debate over the years as to how many of these iterator style modules we should support when the same thing is readily achievable using something like
````
const out = fc.features.map(function (f){
  bboxClip(f, bbox)
})
````
Is there any particular reason why would you use a turf module rather than doing it natively?

Cheers,
Rowan

**Comment by rowanwins** [comment]
yeah I take your point... it's one of those trade off's. I don't have a strong opinion either way, the main downside I see it that it adds another module that needs to be tested and maintained, which might not seem like much but they add up when refactoring has to happen.

**Comment by stevage** [comment]
Yeah, I think you're right. 

Maybe a different approach to this kind of stuff might be having turf add a `.map()` function to an `FeatureCollection` it returns from anything that acts on its `features` property. Probably downsides to that too. It'd solve a perennial problem I have with manipulating GeoJSON and Turf, in which I'm never 100% sure whether my object is a FeatureCollection or an array of Feature. Or whether it's a Feature, or that Feature's properties.
